src/app/dim-api/actions.ts

```diff
@@ -43,17 +43,21 @@ import {
 import { DimApiState } from './reducer';
 import { apiPermissionGrantedSelector, makeProfileKeyFromAccount } from './selectors';
 
-const installApiPermissionObserver = _.once(() => {
+const installApiPermissionObserver = _.once(<D extends Dispatch>(dispatch: D) => {
   // Observe API permission and reflect it into local storage
   // We could also use a thunk action instead of an observer... either way
-  observeStore(
-    (state) => state.dimApi.apiPermissionGranted,
-    (_prev, apiPermissionGranted) => {
-      if (apiPermissionGranted !== null) {
-        // Save the permission preference to local storage
-        localStorage.setItem('dim-api-enabled', apiPermissionGranted ? 'true' : 'false');
-      }
-    },
+  dispatch(
+    observe({
+      id: 'api-permission-observer',
```

Comment:
Does this ID make these observe calls idempotent by default? In other words, can we dispatch `observe` every time instead of having to use a `_.once`?

Reply:
I would need to double check but I think yes

Reply:
@bhollis because this has `runInitially` set it would mean that every invocation would trigger the side effect without the `_.once`. We could change that so the side effect is only run initially if it hasn't already been registered. I think this is more intuitive, so I will add it and ping you again for review.

Reply:
What we really have, is that only one side effect can be registered per id, so adding the same id will overwrite the existing side effect. So I think leaving the `_.once` is probably best in its current state, but we could change the listener to behave slightly differently here. Thoughts?

Reply:
Makes sense to me